src/tools/compiletest/src/runtest.rs

```diff
@@ -1883,18 +1883,85 @@ actual:\n\
         let _ = fs::remove_dir_all(&out_dir);
         self.create_dir_racy(&out_dir);
 
-        let proc_res = self.document(&out_dir);
-        if !proc_res.status.success() {
-            self.fatal_proc_rec("rustdoc failed!", &proc_res);
+        if self.props.check_test_line_numbers_match == true {
+            self.check_rustdoc_test_option();
+        } else {
+            let proc_res = self.document(&out_dir);
```

Comment:
My point from long ago is that this statement should be on the common code path between these two branches. The `proc_res` return value contains the output, and that's what you should match against.